lib/model/folder.go

```diff
@@ -320,7 +327,11 @@ func (f *folder) restartWatch() {
 func (f *folder) startWatch() {
 	ctx, cancel := context.WithCancel(f.ctx)
 	f.model.fmut.RLock()
-	ignores := f.model.folderIgnores[f.folderID]
+	ignores, ok := f.model.folderIgnores[f.folderID]
+	if !ok {
+		l.Debugf("Failed to get ignore patterns for folder %v while starting the filesystem watcher -> aborting, folder will be stopped soon (racy)", f.Description())
+		return
+	}
```

Comment:
Surely, while the parent context is not cancelled, ignores should be there, as the ignores get removed after the folder is stopped (with the <-f.Stopped() stuff).